pkg/util/rankedset/rankedset.go

```diff
@@ -0,0 +1,162 @@
+package rankedset
+
+import "github.com/google/btree"
+
+// Item represents a single object in a RankedSet.
+type Item interface {
+	// Key returns the unique identifier for this item.
+	Key() string
+	// Rank is used to sort items.
+	// Items with the same rank are sorted lexicographically based on Key.
+	Rank() int64
+}
+
+// RankedSet stores Items based on Key (uniqueness) and Rank (sorting).
+type RankedSet struct {
+	rank *btree.BTree
+	set  map[string]*treeItem
+}
+
+// String implements Item using a string.
+// It has two main uses:
+// 1. If all items in a RankedSet are Strings, the set becomes a store of unique strings sorted lexicographically.
+// 2. It serves as a Key item that can be passed into methods that ignore Rank such as RankedSet.Remove.
+type String string
+
+func (s String) Key() string {
+	return string(s)
+}
+
+func (s String) Rank() int64 {
+	return 0
+}
+
+func New() *RankedSet {
+	return &RankedSet{
+		rank: btree.New(32),
+		set:  make(map[string]*treeItem),
+	}
+}
+
+// Add inserts the item into the set.
+// If an item with the same Key existed in the set, it is removed and returned.
+func (s *RankedSet) Add(item Item) Item {
```

Comment:
Make this follow the same naming scheme as k8s sets from `k8s.io/apimachinery/pkg/util/sets`, iow. `Insert`, `Delete`.

Reply:
OK